automation/taskcluster/lib/tasks.py

```diff
@@ -50,109 +98,202 @@ def build_task(self, name, description, command, artifacts={}, scopes=[], featur
+        return self._craft_clean_gradle_task(
+            name='ktlint',
+            description='Running ktlint over all modules',
+            gradle_task='ktlint',
+            treeherder={
+                'jobKind': 'test',
+                'machine': {
+                  'platform': 'lint',
+                },
+                'symbol': 'ktlint',
+                'tier': 1,
+            }
+        )
+
+    def craft_lint_task(self):
+        return self._craft_clean_gradle_task(
+            name='lint',
+            description='Running ktlint over all modules',
+            gradle_task='lint',
+            treeherder={
+                'jobKind': 'test',
+                'machine': {
+                  'platform': 'lint',
+                },
+                'symbol': 'lint',
+                'tier': 1,
             },
-            "provisionerId": 'aws-provisioner-v1',
-            "metadata": {
-                "name": name,
-                "description": description,
-                "owner": self.owner,
-                "source": self.source
+        )
+
+    def _craft_clean_gradle_task(
+        self, name, description, gradle_task, artifacts=None, routes=None, treeherder=None
+    ):
+        return self._craft_build_ish_task(
+            name=name,
+            description=description,
+            command='./gradlew --no-daemon clean {}'.format(gradle_task),
+            artifacts=artifacts,
+            routes=routes,
+            treeherder=treeherder,
+        )
+
+    def craft_compare_locales_task(self):
+        return self._craft_build_ish_task(
+            name='compare-locales',
+            description='Validate strings.xml with compare-locales',
+            command=(
+                'pip install "compare-locales>=5.0.2,<6.0" && '
+                'compare-locales --validate l10n.toml .'
+            ),
+            treeherder={
+                'jobKind': 'test',
+                'machine': {
+                  'platform': 'lint',
+                },
+                'symbol': 'compare-locale',
+                'tier': 2,
             }
+        )
+
+    def _craft_build_ish_task(
+        self, name, description, command, dependencies=None, artifacts=None, scopes=None,
+        routes=None, treeherder=None
+    ):
+        dependencies = [] if dependencies is None else dependencies
+        artifacts = {} if artifacts is None else artifacts
+        scopes = [] if scopes is None else scopes
+        routes = [] if routes is None else routes
+
+        checkout_command = ' && '.join([
+            'export TERM=dumb',
+            'cd .. ',
+            'git clone {}'.format(self.repo_url),
+            'cd reference-browser',
+            'git fetch {} {} --tags'.format(self.repo_url, self.git_ref),
+            'git config advice.detachedHead false',
+            'git checkout {}'.format(self.commit),
+        ])
+
+        command = '{} && {}'.format(checkout_command, command)
```

Comment:
Nit: you could just add `command` to the end of the `.join([])` list :thinking:

Reply:
True! Done.